.coveragerc

```diff
@@ -0,0 +1,33 @@
+[run]
```

Comment:
What is this file for? Could you please describe here as comments (or put links where to read)

Reply:
To configure coverage. For example, don't include into the coverage lines like __name__ == "__main__", which won't be passed in tests anyway. Try run_coverage and see the results in htmlcov folder, which will be created. Ideally the coverage should be 100%. As to me, I always (almost) follow test driven development methodology: start new function coding with tests, than add and develop the function itself. It's much more productive way for many reasons: you have to obey proper design, maintain integrity through any changes.  
Your code has grown up much above a use-once-and-throw-away utility and deserves this attitude.

Reply:
Thanks. I suppose this is for [coverage](https://coverage.readthedocs.io/en/coverage-4.4.2/index.html) that one should run locally. Do I right understand, that here it is used to estimate the amount of code that is run during the tests?

Reply:
After running coverage you could see the complete report (by default, in htmlcov directory) on the lines of the code visited on test runs. Yes, it also reports percentage of code coverage and range of lines missed on console.